crates/data_components/src/databricks/sql_warehouse.rs

```diff
@@ -0,0 +1,634 @@
+        Ok(Self { pool })
+    }
+}
+
+struct SqlWarehouseConnectionPool {
+    api: Arc<SqlWarehouseApi>,
+}
+
+#[async_trait]
+impl DbConnectionPool<Arc<SqlWarehouseApi>, &'static dyn Sync> for SqlWarehouseConnectionPool {
+    async fn connect(
+        &self,
+    ) -> Result<
+        Box<dyn DbConnection<Arc<SqlWarehouseApi>, &'static dyn Sync>>,
+        Box<dyn std::error::Error + Send + Sync>,
+    > {
+        Ok(Box::new(SqlWarehouseConnection {
+            api: Arc::clone(&self.api),
+        }))
+    }
+
+    fn join_push_down(&self) -> JoinPushDown {
+        JoinPushDown::Disallow
+    }
+}
+
+struct SqlWarehouseApi {
+    client: Client,
+    host: String,
+    sql_warehouse_id: String,
+    token_provider: Arc<dyn TokenProvider>,
+}
+
+impl SqlWarehouseApi {
+    fn new(
+        host: &str,
+        sql_warehouse_id: &str,
+        token_provider: Arc<dyn TokenProvider>,
+    ) -> Result<Self, Error> {
+        let client = ClientBuilder::new()
+            .user_agent(super::user_agent())
+            .build()
+            .context(ClientBuildFailedSnafu)?;
+
+        Ok(Self {
+            client,
+            host: host.to_string(),
+            sql_warehouse_id: sql_warehouse_id.to_string(),
+            token_provider,
+        })
+    }
+
+    async fn get_schema(&self, table: &TableReference) -> Result<SchemaRef, Error> {
+        let token = self.token_provider.get_token();
+        let sql = format!("DESCRIBE TABLE {table}");
+        let payload = self.create_schema_payload(table, &sql)?;
+
+        let response = self.execute_request(&token, &payload).await?;
+        schema_from_json(&response)
+    }
+
+    fn create_schema_payload(&self, table: &TableReference, sql: &str) -> Result<Value, Error> {
+        Ok(json!({
+            "warehouse_id": self.sql_warehouse_id,
+            "catalog": table.catalog().ok_or_else(|| Error::FullyQualifiedPath{ reason: "missing catalog".into() })?,
+            "schema": table.schema().ok_or_else(|| Error::FullyQualifiedPath{ reason: "missing schema".into() })?,
+            "statement": sql,
+        }))
+    }
+
+    async fn execute_request(&self, token: &str, payload: &Value) -> Result<Value, Error> {
+        let url = format!("https://{}/api/2.0/sql/statements/", self.host);
+        self.client
+            .post(&url)
+            .bearer_auth(token)
+            .json(payload)
+            .send()
+            .await
+            .context(HttpRequestFailedSnafu)?
+            .json()
+            .await
+            .context(JsonParsingFailedSnafu)
+    }
+
+    // Fetch the arrow data at the external links, repeating for each chunk
+    async fn fetch_external_links(
+        &self,
+        result_object: Value,
+    ) -> Result<SendableRecordBatchStream, Error> {
+        let mut streams = Vec::new();
+        let token = self.token_provider.get_token();
+        let mut external_link = Self::extract_external_links(result_object)?;
+        loop {
```

Comment:
We should update this into real streamed approach as it seems right now we collect all batches first and start streaming only after that

This can be done next